InAppUtils/InAppUtils.m

```diff
@@ -117,42 +161,35 @@ - (void) doPurchaseProduct:(NSString *)productIdentifier
                 payment.applicationUsername = username;
             }
             [[SKPaymentQueue defaultQueue] addPayment:payment];
-            _callbacks[RCTKeyForInstance(payment.productIdentifier)] = callback;
         } else {
-            callback(@[@"invalid_product"]);
+            NSLog(@"invalid_product"); // TODO we lose feedback here, how do we solve this? Send error event?
```

Comment:
methode kan als een promise werken lijkt me, en bij een error oid rejecten. Dan kunnen we het in JS afhandelen met een logger.error